src/extension/wps/wps-core/src/main/java/org/geoserver/wps/resource/DefaultProcessArtifactsStore.java

```diff
@@ -42,13 +40,6 @@ public Resource getArtifact(String executionId, ArtifactType type, String name)
         } else {
             result = store.get(Paths.path(executionId, "tmp", name));
         }
-        if (result != null) {
-            // make sure we have the root directory
-            File dir = result.parent().dir();
-            if (!dir.exists()) {
-                dir.mkdirs();
-            }
-        }
```

Comment:
See one of my previous comments, directories in the ResourceStore are created on-the-fly.